### PR TITLE
Fix package security vulnerabilities

### PR DESCRIPTION
**Pull request by dannyroberts**
Eventually we should be able to get rid of `uninstall-requirements.txt` as well, and replace that process with `pip-sync`. Leaving that for another PR.

### CODE REVIEW

**Review by snopoke — APPROVED** [review]

**Comment by emord** [comment]
@dannyroberts I already started this https://github.com/dimagi/commcare-hq/pull/21230 https://github.com/dimagi/commcare-hq/pull/21229

File: requirements/requirements.in [hunk 1]
```diff
@@ -22,7 +22,7 @@ djangorestframework==3.5.4
 elasticsearch==1.9.0
 eulxml==1.1.3
 ghdiff==0.4
-gunicorn==19.4
+gunicorn
```
Comment (emord) [inline requirements/requirements.in]: Why don't these have version numbers?

**Comment by dannyroberts** [comment]
Oh, awesome! Sorry I missed that. Looks like my [recent requirements pinning changes](https://github.com/dimagi/commcare-hq/pull/21251) broke your PRs.

Should we just keep this PR which works off that? It's nice to see you already tested gunicorn on staging for a couple weeks; that'll make getting this in a lot quicker than I thought it would be!

File: requirements/requirements.in (hunk 1 above)
Comment (dannyroberts) [reply requirements/requirements.in]: Yeah good question. This is sort of the crux of the new system. The `*requirements.in` files contain "logical" dependencies, and then the `*requirements.txt` contain the _actual_, frozen, dependencies. This lets us later run `make requirements`, it'll update to the latest version of the package in `*requirements.txt`, and we can then choose how and when to commit those. (A compromise option is to at that time add back a max version like `gunicorn<20` if we want to postpone a particular update.)

Does that make sense? We'll talk a bunch more about this in the session on Wednesday, including what the actual process around updating dependencies will be (run on a schedule? automated PRs? etc.)

Comment (emord) [reply requirements/requirements.in]: Cool. I'll wait til Wednesday. Was just confused since all the others have version info

**Review by emord — APPROVED** [review]
pycrypto changes are a bit scary (their docs said use MODE_EAX by default for AES which is why it's wrong in my other PR). Do you know if the other usages are fully tested?

**Comment by dannyroberts** [comment]
Yeah good point. There are places in the code where something is imported from `Crypto`, so I'll make sure each of them gets tested.

**Review by biyeun — APPROVED** [review]

**Comment by dannyroberts** [comment]
@emord I made a couple attempts at writing some tests for these uses, but found it surprisingly difficult (`sign` for example doesn't create the same output for the same input, so the simplest kinds of tests don't work). I'm sure I could figure this out with a bit more time, but this has fallen pretty low on my list, so I decided to just audit the half-dozen usages manually and look them up in the changelog to see if there are breaking changes, and the only one that came up is the one we already fixed. I'm satisfied with that, so ready to go if you're okay with it.

**Pull request merged**